modules/core/src/main/scala/fs2/kafka/consumer/KafkaConsumeChunk.scala

```diff
@@ -0,0 +1,96 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package fs2.kafka.consumer
+
+import cats.effect.Concurrent
+import cats.syntax.flatMap.*
+import cats.Monad
+import fs2.*
+import fs2.kafka.consumer.KafkaConsumeChunk.CommitNow
+import fs2.kafka.CommittableConsumerRecord
+import fs2.kafka.CommittableOffsetBatch
+import fs2.kafka.ConsumerRecord
+
+trait KafkaConsumeChunk[F[_], K, V] extends KafkaConsume[F, K, V] {
+
+  /**
+    * Consume from all assigned partitions concurrently, processing the records in `Chunk`s. For
+    * each `Chunk`, the provided `processor` is called, after that has finished the offsets for all
+    * messages in the chunk are committed.<br><br>
+    *
+    * This method is intended to be used in cases that require at-least-once-delivery, where
+    * messages have to be processed before offsets are committed. By relying on the methods like
+    * [[partitionedStream]], [[records]], and similar, you have to correctly implement not only your
+    * processing logic but also the correct mechanism for committing offsets. This can be tricky to
+    * do in a correct and efficient way.<br><br>
+    *
+    * Working with `Chunk`s of records has several benefits:<br>
+    *   - As a user, you don't have to care about committing offsets correctly. You can focus on
+    *     implementing your business logic<br>
+    *   - It's very straightforward to batch several messages from a `Chunk` together, e.g. for
+    *     efficient writes to a persistent storage<br>
+    *   - You can liberally use logic that involves concurrency, filtering, and re-ordering of
+    *     messages without having to worry about incorrect offset commits<br>
+    *
+    * <br>
+    *
+    * The `processor` is a function that takes a `Chunk[ConsumerRecord[K, V]]` and returns a
+    * `F[CommitNow]`. [[CommitNow]] is isomorphic to `Unit`, but helps in transporting the intention
+    * that processing of a `Chunk` is done, offsets should be committed, and no important processing
+    * should be done afterwards.<br><br>
+    *
+    * The returned value has the type `F[Nothing]`, because it's a never-ending process that doesn't
+    * terminate, and therefore doesn't return a result.
+    *
+    * @note
+    *   This method does not make any use of Kafka's auto-commit feature, it implements "manual"
+    *   commits in a way that suits most of the common use cases.
+    * @note
+    *   you have to first use `subscribe` or `assign` the consumer before using this `Stream`. If
+    *   you forgot to subscribe, there will be a [[NotSubscribedException]] raised in the `Stream`.
+    * @see
+    *   [[partitionedStream]]
+    * @see
+    *   [[CommitNow]]
+    */
+  final def consumeChunk(
+    processor: Chunk[ConsumerRecord[K, V]] => F[CommitNow]
+  )(implicit F: Concurrent[F]): F[Nothing] = partitionedStream
+    .map(
+      _.chunks.evalMap(consume(processor))
+    )
+    .parJoinUnbounded
+    .compile
+    .drain
+    .flatMap[Nothing](_ => F.never)
```

Comment:
I'm not sure if this is the right way to solve the compile error. I think the ideal solution would be making the `_` type parameter of the `F` in `KafkaConsumeChunk`  covariant, but that would require it to be covariant in `KafkaConsumer` as well, which would probably rather invasive

Reply:
I was going to propose:

```scala
    .parJoinUnbounded
    .drain
    .compile
    .lastOrError
```

WDYT?

Reply:
But I can't test it due to compilation issues with the tests.

Reply:
I'm not sure if the types will line up with `lastOrError`, but I'll give it a try. I'm currently working on fixing the compilation issues in the tests (I only tried it locally with Scala 3 and didn't think about different Collection APIs in the 2.x versions)

Reply:
The tests should be fixed, compiles on my machine :tm: with 2.12.18, 2.13.10, and 3.3.0.

I tried `lastOrError`, but I think this leads to the same issue with the initial `>> F.never` variant

Reply:
Technically, `.drain >>[Nothing] F.never` would work as well, but with `flatMap` I think it's a bit less confusing syntax-wise

Reply:
>I tried lastOrError, but I think this leads to the same issue with the initial >> F.never variant

Can you share the code? I've tried it and works (compiles and tests pass) 🤔

Reply:
You're right, sorry! I missed the `drain` before `compile`.

I like this more than what's currently implemented, because it addresses the point I brought up earlier: https://github.com/fd4s/fs2-kafka/pull/1281#issuecomment-1849729779 If for some reason the stream terminates, the whole action ends. With `F.never`, it "ends" silently.

We could bikeshed if it's `onlyOrError` or `lastOrError` now, but I think that doesn't matter

Reply:
> We could bikeshed if it's `onlyOrError` or `lastOrError` now, but I think that doesn't matter

Exactly! I'm also uncertain of what to use (`only` vs `last`), so whatever you choose will be good :)

Reply:
I went with `only`, because `last` has the notion of "drop some elements" and in that case `only` is closer to "there won't be any elements in the stream"